src/components/toast/components/close-button/types.ts

```diff
@@ -0,0 +1,14 @@
+export interface CloseButtonProps {
+  /**
+   * A function that will be called when the button is clicked.
+   */
+  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+  /**
+   * Optional class to be added to the button element.
+   */
+  className?: string
+  /**
+   * A label that describes the buttons action. Applied as aria-label value.
+   */
+  ariaLabel: string
```

Comment:
Note that `CloseButton` as well as this types file is mostly cribbed from https://github.com/hashicorp/react-components/tree/main/packages/close-button - with minor style adjustments for dev-dot, and a simplified props API that cuts out what we have no concrete use case for (before June 20, at least).